server/server.go

```diff
@@ -691,19 +511,43 @@ func (s *Server) spawn(l net.Listener, c net.Conn) {
 		tconn.Close()
 		return
 	}
-	handle := client.SpawnConn(conn, s.config.maxConnPendingRequests)
-	s.listeners[l][handle] = struct{}{}
+
+	s.listeners[l][tconn] = struct{}{}
 	s.mtx.Unlock()
-	log.Debugf("%s: spawned", connStr)
+	log.Debugf("%s: serving", connStr)
+	handler := &handler{
+		name:        connStr,
+		outstanding: s.config.maxConnPendingRequests,
+		s:           s,
+		c:           nil,
+		limited:     limited,
+		conn:        tconn,
+		listener:    l,
+	}
+	handler.c = sync.NewCond(&handler.mtx)
+
+	for {
+		pkt := new(protocol.Packet)
+		handler.mtx.Lock()
+		for handler.outstanding == 0 {
```

Comment:
By similar logic as above, shouldn't this check if `handler.outstanding > maxConnPendingRequests`, i.e, check that we are not trying to process more than `maxConnPendingRequests` at the same time.

Reply:
Same reply as a above. I start by filling the bucket of tokens to the max.